lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/dllmain.cpp

```diff
@@ -1244,6 +1257,37 @@ void eos_create(EOSConfig& eosConfig)
         steam_platform.SteamMajorVersion = eos_steam_config.steamSDKMajorVersion;
         steam_platform.SteamMinorVersion = eos_steam_config.steamSDKMinorVersion;
 
+        // For each element in the array (each of which is a string of an api version information)
+        // iterate across each character, and at the end of a string add a null terminator \0
+        // then add one more null terminator at the end of the array
+        std::vector<char> steamApiInterfaceVersionsAsCharArray = std::vector<char>();
+
+        // steam_platform needs to have a count of how many bytes the "array" is, stored in SteamApiInterfaceVersionsArrayBytes
+        // This has some fuzzy behavior; if you set it to 0 or count it up properly, there won't be a logged problem
+        // if you put a non-zero amount that is insufficient, there will be an unclear logged error message
+        // HACK: For now the right method seems to be counting each char added to the array, including the null terminator
+        uint32_t totalBytes = 0;
+
+        for (int apiInterfaceVersionIndex = 0; apiInterfaceVersionIndex < eos_steam_config.steamApiInterfaceVersionsArray.size(); apiInterfaceVersionIndex++)
+        {
+            std::string currentFullValue = eos_steam_config.steamApiInterfaceVersionsArray[apiInterfaceVersionIndex];
+
+            for (int characterIndex = 0; characterIndex < currentFullValue.length(); characterIndex++)
+            {
+                char currentCharacter = currentFullValue[characterIndex];
+                steamApiInterfaceVersionsAsCharArray.push_back(currentCharacter);
+                totalBytes++;
+            }
+
+            steamApiInterfaceVersionsAsCharArray.push_back('\0');
+            totalBytes++;
+        }
+        steamApiInterfaceVersionsAsCharArray.push_back('\0');
```

Comment:
I made this go through the motions and then use `.size()`, and that did indeed get the correct size! I had been using `sizeof`, and that was only counting up until the first `\0`. `size()` is functioning correctly, so we'll use that instead of counting up in the loop like this.